SimTracker/TrackTriggerAssociation/src/StubAssociation.cc

```diff
@@ -60,7 +60,7 @@ namespace tt {
     // Get all TPs that are matched to these stubs in at least 'tpMinLayers' layers and 'tpMinLayersPS' ps layers
     vector<TPPtr> tpPtrs = associate(ttStubRefs);
     // remove TPs with more then 'tpMaxBadStubs2S' not associated 2S stubs and more then 'tpMaxBadStubsPS' not associated PS stubs
-    auto check = [this, ttStubRefs](const TPPtr& tpPtr) {
+    auto check = [this, &ttStubRefs](const TPPtr& tpPtr) {
```

Comment:
why is this change here? we need to be VERY careful about changing in the stub or MC truth associator classes.

Reply:
my point is that - any change to the association classes needs to happen in a dedicated separate PR which motivated change. if this is a mistake, please fix it so that there are no changes in the association package for this PR.

Reply:
looking at this further I see that this is not the regular associator class (TTStubAssociator) but another stub associator class that is somehow specific to the new KF code?? why is a separate stub associator class needed in the first place? in this case though it's fine if this is coupled I guess since it then shouldn't affect anything which is currently being used by the L1 tracking.

Reply:
As far as I can see we both don't use the TTStubAssociator. We use TTClusterAssociator to create something to associate stub collections with truth. Since I have several analyzer I would have to create this same something several times. Therefore I created this class to do it only once in order to save CPU time.

Reply:
I think this is OK. Thomas's StubAssociation class does have some nice features not present in the standard associators. But as we've observed when it was added, it's ugly that there is some code duplication.